File: rdvp/src/main/java/com/grdvp/service/interfaces/PatientServiceImpl.java
package com.grdvp.service.interfaces;

import com.grdvp.entity.Patient;
import java.time.LocalDate;
import java.util.List;

public interface PatientServiceImpl {
    void addPatient(Patient patient);
    void addPersonalInformation(Patient patient);
    void addMedicalHistory(Patient patient, List<String> medicalHistory);
    String generatePatientCode();
    Patient connexion(String email, String password);
    void completePatientInfo(Patient patient, LocalDate dateNaissance, String adresse);
    void addAntecedent(Patient patient, String antecedent);
    List<Patient> getAllPatien();
}

File: src/App/service/interface/PatientServiceImpl.php
<?php

namespace App\service\interface;

use App\entity\Patient;
use DateTime;

interface PatientServiceImpl
{
    public function addPatient(Patient $patient): void;

    public function addPersonnalInformation(Patient $patient): void;

    public function addMedicalHistory(Patient $patient, string $medicalHistory): void;

    public function generatePatientCode(): string;
    
    public function connexion(string $email, string $password): ?Patient;
    
    public function completePatientInfo(Patient $patient, DateTime $dateNaissance, string $adresse): void;

    public function addAntecedent(Patient $patient, string $antecedent): void;
    
    public function getConnectedPatientInfo(int $patientId): Patient;

    public function getAllPatien(): array;
}